canopen/objectdictionary/__init__.py

```diff
@@ -285,6 +285,9 @@ def decode_raw(self, data):
             return data.decode("utf_16_le")
         elif self.data_type in self.STRUCT_TYPES:
             try:
+                needed_bytes = self.STRUCT_TYPES[self.data_type].size
+                if needed_bytes < len(data):
```

Comment:
A warning should be logged so if there is a mismatch between Object Dictionary and actual implementation and the user has issues with faulty values he/she might get an idea where to start looking.

Reply:
Should that be a `logger.debug()` then, as it would appear for every received PDO, possibly affecting performance?

Reply:
I think I changed my mind. It’s probably better to make it configurable with default to enforce the correct size. An informative exception can be raised which explains how to configure the node to accept larger data packets. What do you think?

Reply:
The idea sounds nice, but where in the class stack should this be implemented?  Naming suggestions? Something with `strict` in the name probably...